packages/graphql-language-service-server/src/GraphQLCache.ts

```diff
@@ -54,8 +54,15 @@ const {
 
 export async function getGraphQLCache(
   configDir: Uri,
+  extensions?: Array<(config: GraphQLConfig) => GraphQLConfig>,
 ): Promise<GraphQLCacheInterface> {
-  const graphQLConfig = await loadConfig({ rootDir: configDir });
+  let graphQLConfig = await loadConfig({ rootDir: configDir });
+  if (extensions && extensions.length > 0) {
+    /* eslint-disable no-await-in-loop */
+    for (const extension of extensions) {
+      graphQLConfig = await extension(graphQLConfig);
```

Comment:
@divyenduz so this is intended to re-declare `graphQLConfig` on every invocation? I'm guessing the result of `await extension(graphQLConfig)` returns the previous config plus the extension?

Reply:
Yes, as of now, the signature of an extension is a function that takes a GraphQLConfig as input and returns a GraphQL config as its output with the modification (extension) applied.

`extensions?: Array<(config: GraphQLConfig) => GraphQLConfig>`
Which implies, `extension: (config: GraphQLConfig) => GraphQLConfig`